File: le12s-v2.5.0.vb
module module1
	dim a, b, c as integer
	Sub main
		console.writeline("Welcome to Muhammad Sreywel's quadratic equation solver [version 2.5.0]")
		console.writeline("Please make sure that the equation you want to solve is in the form { ax^2 + bx + c = 0 } before you start.")
		console.writeline("press enter to start...".toupper)
		console.readline()
		solve
	end sub
	Sub solve
		Dim ans As String=""
		Do
			Try
				console.write("a: ")
				a=console.readline()
				console.write("b: ")
				b=console.readline()
				console.write("c: ")
				c=console.readline()
			Catch n As exception
				console.writeline("Please use decimal numbers only..")
				solve
			End Try
			Dim d As Single = (b*b)-(4*a*c)
			If a=0
				console.write("x = " & CStr(-c/b))
			ElseIf d=0
				console.write("x = " & CStr(-b/2/a))
			ElseIf d<0
				console.writeline("x1 = "+CStr((math.Sqrt(math.abs(d))-b)/(2*a))+"i")
				console.writeline("x2 = "+CStr((-b-math.sqrt(math.abs(d)))/(2*a))+"i")
			Else
				console.writeline("x1 = "+CStr((math.Sqrt(d)-b)/(2*a)))
				console.writeline("x2 = "+CStr((-b-math.sqrt(d))/(2*a)))
			End If
			console.writeline()
			console.write("Do you want to solve another equation?(y/n)")
			ans = console.readline()
		Loop While ans.tolower="y"
	end sub
end module

File: le12s-v2.5.0.cs
using System;
class module1
{
	static void Main(string[] args)
	{
		Console.WriteLine("Welcome to Muhammad Sreywel's quadratic equation solver [version 2.5.0]");
		Console.WriteLine("Please make sure that the equation you want to solve is in the form { ax^2 + bx + c = 0 } before you start.");
		Console.WriteLine("press enter to start...".ToUpper());
		Console.ReadLine();
		solve();
	}
	static void solve()
	{
		int a=0, b=0, c=0;
		string ans = "";
		do
		{
			try
			{
				Console.Write("a: ");
				a=int.Parse(Console.ReadLine());
				Console.Write("b: ");
				b=int.Parse(Console.ReadLine());
				Console.Write("c: ");
				c=int.Parse(Console.ReadLine());
			}
			catch
			{
				Console.WriteLine("Please use decimal numbers only..");
				solve();
			}
			float d = (b*b)-(4*a*c);
			if(a==0){
				Console.Write("x = " + (-c/b));}
			else{if(d==0){
				Console.Write("x = " + (-b/2/a));}
			else{if(d<0){
				Console.WriteLine("x1 = "+((Math.Sqrt(Math.Abs(d))-b)/(2*a))+"i");
				Console.WriteLine("x2 = "+((-b-Math.Sqrt(Math.Abs(d)))/(2*a))+"i");}
			else{
				Console.WriteLine("x1 = "+((Math.Sqrt(d)-b)/(2*a)));
				Console.WriteLine("x2 = "+((-b-Math.Sqrt(d))/(2*a)));
			}}}
			Console.WriteLine();
			Console.Write("Do you want to solve another equation?(y/n)");
			ans = Console.ReadLine();
		}while(ans.ToLower()=="y");
	}
}